### PR TITLE
ocamlPackages.merlin: 3.3.6 -> 3.3.8

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
https://github.com/ocaml/merlin/releases/tag/v3.3.7

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/ocaml/merlin/default.nix [hunk 1]
```diff
@@ -2,13 +2,15 @@
 
 buildDunePackage rec {
   pname = "merlin";
-  version = "3.3.6";
+  version = "3.3.7";
 
   minimumOCamlVersion = "4.02.1";
 
+  useDune2 = true;
```
Comment (vbgl) [inline pkgs/development/tools/ocaml/merlin/default.nix:9]: This change is not expected, is it? It might be worth reporting the issue upstream…

Comment (marsam) [reply pkgs/development/tools/ocaml/merlin/default.nix:9]: Reported upstream here https://github.com/ocaml/merlin/issues/1157

**Comment by vbgl** [comment]
Thanks. Let’s directly jump to version 3.3.8 then!

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.merlin

**Pull request merged**